packages/editor/src/components/block-list/block.js

```diff
@@ -677,14 +680,19 @@ const applyWithSelect = withSelect( ( select, { clientId, rootClientId, isLargeV
 		mode: getBlockMode( clientId ),
 		isSelectionEnabled: isSelectionEnabled(),
 		initialPosition: getSelectedBlocksInitialCaretPosition(),
-		isEmptyDefaultBlock: block && isUnmodifiedDefaultBlock( block ),
-		isPreviousBlockADefaultEmptyBlock: previousBlock && isUnmodifiedDefaultBlock( previousBlock ),
+		isEmptyDefaultBlock: blockName && isUnmodifiedDefaultBlock( { name: blockName, attributes: blockAttributes } ),
+		isPreviousBlockADefaultEmptyBlock: previousBlockClientId && isUnmodifiedDefaultBlock( {
```

Comment:
We should never want to construct the object of a block. It seems we could use `createBlock`?

Reply:
This is not really returned as a prop though, we just return a boolean.

Reply:
I'm not concerned so much about what we expose to the component as a prop, just limiting the exposure to _any code_ which makes an assumption about the block abstraction, should that abstraction come to change in the future.

Reply:
yes, I don't like `isUnmodifiedDefaultBlock` either, I think I'd be fine with `createBlock` for now.